ladybug_geometry/geometry3d/mesh.py

```diff
@@ -392,6 +433,16 @@ def _mesh_scale(self, verts, factor):
         _new_mesh._vertex_normals = self._vertex_normals
         return _new_mesh
 
+    def _check_vertices_input(self, vertices):
+        """Check the input vertices."""
+        if not isinstance(vertices, tuple):
+            vertices = tuple(vertices)
+        for vert in vertices:
+            assert isinstance(vert, Point3D), \
+                'Expected Point3D for {} vertex. Got {}.'.format(
+                    self.__class__.__name__, type(vert))
+        self._vertices = vertices
```

Comment:
This is more of a design taste but I think it is cleaner if all the assignments happen under `__init__` method.

```python
def __init__(self, vertices):
    self._vertices = self._check_vertices_input(vertices)
    # ...

def _check_vertices_input(self, vertices):
    # ...
    return vertices
```

Reply:
Good point.  I went back and forth with myself about whether it was better to have a duplicated line of code across all of the classes that inherit this method.  But, now that you say it, I agree it should be in `__init__` and I can see from an outsiders perspective that having it in `__init__` is much clearer.  I'll change it now.

Reply:
> whether it was better to have a duplicated line of code across all of the classes that inherit this method

Is it because you don't call the `__init__` method from subclasses? You should just call the baseclass `__init__` and that should take care of it. You only need it once in all your code.